socorro/processor/breakpad_transform_rules.py

```diff
@@ -459,3 +491,153 @@ def _predicate(
+
+    #--------------------------------------------------------------------------
+    def _execute_external_process(self, command_line, processor_meta):
+        stackwalker_output, return_code = super(
+            BreakpadStackwalkerRule2015,
+            self
+        )._execute_external_process(command_line, processor_meta)
+
+        if not isinstance(stackwalker_output, Mapping):
+            processor_notes.append(
+                "MDSW produced unexpected output: %s..." %
+                str(stackwalker_output)[:10]
+            )
+            stackwalker_output = {}
+
+        stackwalker_data = DotDict()
+        stackwalker_data.json_dump = stackwalker_output
+        stackwalker_data.mdsw_return_code = return_code
+
+        stackwalker_data.mdsw_status_string = stackwalker_output.get(
+            'status',
+            'unknown error'
+        )
+        stackwalker_data.success = stackwalker_data.mdsw_status_string == 'OK'
+
+        if return_code == 124:
+            processor_meta.processor_notes.append(
+                "MDSW terminated with SIGKILL due to timeout"
+            )
+        elif return_code != 0 or not stackwalker_data.success:
+            processor_meta.processor_notes.append(
+                "MDSW failed on '%s': %s" % (
+                    command_line,
+                    stackwalker_data.mdsw_status_string
+                )
+            )
+
+        return stackwalker_data, return_code
+
+    #--------------------------------------------------------------------------
+    def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
+        if 'additional_minidumps' not in processed_crash:
+            processed_crash.additional_minidumps = []
+        with self._temp_raw_crash_json_file(
+            raw_crash,
+            raw_crash.uuid
+        ) as raw_crash_pathname:
+            for dump_name, dump_pathname in raw_dumps.iteritems():
+
+                if processor_meta.quit_check:
+                    processor_meta.quit_check()
+
+                # this rule is only interested in dumps targeted for the
+                # minidump stackwalker external program.  As of the writing
+                # of this code, there is one other dump type.  The only way
+                # to differentiate these dump types is by the name of the
+                # dump.  All minidumps targeted for the stackwalker will have
+                # a name with a prefix specified in configuration:
+                if not dump_name.startswith(self.config.dump_field):
+                    # dumps not intended for the stackwalker are ignored
+                    continue
+
+                if self.config.chatty:
+                    self.config.logger.debug(
+                        "BreakpadStackwalkerRule: %s, %s",
+                        dump_name,
+                        dump_pathname
+                    )
+
+
+                command_parameters = dict(self.config)
```

Comment:
Is it deliberate to make a copy here? A much neater syntax to achieve this is to make it on the fly. 
```python
command_line = self.config.command_line.format(
    **dict(
        self.config, 
        dump_file_pathname=dump_pathname,
        raw_crash_pathname=raw_crash_pathname
    )
)
```